src/codegen/dyngen.rs

```diff
@@ -125,24 +136,50 @@ impl DynamicItems {
             assert_eq!(args.len(), args_identifiers.len());
         }
 
-        self.struct_members.push(quote! {
-            pub #ident: Result<unsafe extern #abi fn ( #( #args ),* ) #ret, ::libloading::Error>,
-        });
+        self.has_required |= is_required;
+
+        self.struct_members.push(
+            if is_required {
+                quote! {
+                    pub #ident: unsafe extern #abi fn ( #( #args),* ) #ret,
+                }
+            } else {
+                quote! {
+                    pub #ident: Result<unsafe extern #abi fn ( #( #args ),* ) #ret, ::libloading::Error>,
+                }
+            }
+        );
 
         // We can't implement variadic functions from C easily, so we allow to
         // access the function pointer so that the user can call it just fine.
         if !is_variadic {
-            self.struct_implementation.push(quote! {
-                pub unsafe fn #ident ( &self, #( #args ),* ) -> #ret_ty {
-                    let sym = self.#ident.as_ref().expect("Expected function, got error.");
-                    (sym)(#( #args_identifiers ),*)
+            self.struct_implementation.push(
+                if is_required {
+                    quote! {
+                        pub unsafe fn #ident ( &self, #( #args ),* ) -> #ret_ty {
```

Comment:
Same here, at least the method signature could be de-duplicated, and if you do something like `let sym = self.#ident`, the function call as well.

Reply:
👍 - I pulled the body of the function into a separate `TokenStream`.